flow/connectors/core.go

```diff
@@ -25,6 +25,7 @@ var ErrUnsupportedFunctionality = errors.New("requested connector does not suppo
 type Connector interface {
 	Close() error
 	ConnectionActive(context.Context) error
+	ValidateCheck(context.Context) error
```

Comment:
Instead of stub methods you can make something like `GetTableSchemaConnector` then in validation check if you can cast connector to `ValidationConnector`